Mathlib/NumberTheory/LegendreSymbol/GaussEisensteinLemmas.lean

```diff
@@ -83,7 +83,7 @@ private theorem gauss_lemma_aux₁ (p : ℕ) [Fact p.Prime] {a : ℤ}
           (∏ x in Ico 1 (p / 2).succ, if (a * x : ZMod p).val ≤ p / 2 then (1 : ZMod p) else -1) =
           ∏ x in (Ico 1 (p / 2).succ).filter fun x : ℕ => ¬(a * x : ZMod p).val ≤ p / 2, -1 :=
         prod_bij_ne_one (fun x _ _ => x)
-          (fun x => by split_ifs <;> simp_all (config := { contextual := true }))
+          (fun x => by split_ifs <;> (dsimp; simp_all))
```

Comment:
Why did this change?

Reply:
The simp call in `split_ifs` became slightly more powerful, and left us in a different state. That then resulted in `simp_all` taking a wrong turn, which can be averted by making sure `dsimp` gets a chance to beta reduce first.

Reply:
That suggests to me that we have a confluence issue somewhere, but I guess fixing it is not in scope here.